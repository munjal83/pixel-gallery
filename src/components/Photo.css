.container {
    display: flex;
    flex-wrap: nowrap;
    justify-content: space-between;
    overflow: hidden;
    background-color: grey;
    margin: 0 auto;
}
