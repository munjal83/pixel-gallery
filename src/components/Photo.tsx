import React from 'react';

type PhotoProps = {
    data: any,
}

class Photo extends React.Component<{}, PhotoProps> {
	constructor(props: any) {
        super(props);
      }

	componentDidMount() {
		
	}

	render() {
	
		return (
			<div>
				
			</div>
		)
	}
} 

export default Photo;
