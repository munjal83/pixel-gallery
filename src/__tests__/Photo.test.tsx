import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Photo from '../components/Photo';

const photo = {
  image_url: "",
  times_viewed: 999,
  user: {
    fullname: "john doe"
  }
}

describe('Photo', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Photo selectedPhoto={photo} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches the snapshot', () => {
    const tree = renderer.create(<Photo selectedPhoto={photo}/>).toJSON();
    expect(tree).toMatchSnapshot();
});

  it('renders Photo component', () => {
    render(<Photo selectedPhoto={photo} />);

    expect(screen.getByText(/by/)).toBeInTheDocument();
  });

  it('renders with props', () => {
    render(<Photo selectedPhoto={photo} />);

    expect(screen.getByText("Views: 999")).toBeInTheDocument();
  })
});
