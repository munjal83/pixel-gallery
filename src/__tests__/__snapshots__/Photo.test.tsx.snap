// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photo matches the snapshot 1`] = `
<div
  className="container"
>
  <ul
    className="MuiGridList-root-48"
    style={
      Object {
        "height": "100vh",
        "margin": -2,
        "padding": "0 auto",
        "width": "100vw",
      }
    }
  >
    <li
      className="MuiGridListTile-root-49"
      style={
        Object {
          "height": 604,
          "padding": 2,
          "width": "50%",
        }
      }
    >
      <div
        className="MuiGridListTile-tile-50"
      >
        <img
          alt=""
          src=""
        />
        <div
          className="MuiGridListTileBar-root-53 MuiGridListTileBar-titlePositionTop-55"
          style={
            Object {
              "background": "linear-gradient(to bottom, rgba(0,0,0,0.7) 0%, rgba(0,0,0,0.3) 70%, rgba(0,0,0,0) 100%)",
            }
          }
        >
          <div
            className="MuiGridListTileBar-titleWrap-57"
          >
            <div
              className="MuiGridListTileBar-title-60"
            >
              Views: 999
            </div>
          </div>
        </div>
        <div
          className="MuiGridListTileBar-root-53 MuiGridListTileBar-titlePositionBottom-54 MuiGridListTileBar-rootSubtitle-56"
        >
          <div
            className="MuiGridListTileBar-titleWrap-57"
          >
            <div
              className="MuiGridListTileBar-title-60"
            />
            <div
              className="MuiGridListTileBar-subtitle-61"
            >
              <span>
                by: 
                john doe
              </span>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      className="MuiGridListTile-root-49"
      style={
        Object {
          "height": "auto",
          "padding": 2,
          "width": "50%",
        }
      }
    >
      <div
        className="MuiGridListTile-tile-50"
      >
        <h1
          className="MuiTypography-root-64 MuiTypography-display1-68 MuiTypography-gutterBottom-81"
        />
      </div>
    </li>
  </ul>
</div>
`;
